.github/workflows/unit_test.yml

```diff
@@ -0,0 +1,30 @@
+name: Unit test
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install bleak
+      - name: Test with pytest
+        run: |
+          pip install pytest
+          tries=0
+          until [ "$tries" -ge 5 ]
+          do
```

Comment:
Why does it take 5 tries?

Reply:
It's allowed to take 5 tries, but if `pytest` is successful the `&& break` will break out of this loop. The loop ensures that if pytest fails unexpectedly, it is allowed to retry. I guess this design is more common for tests that rely on a third party API/service which might not work 100%.

Reply:
I think we can drop the 5 tries. If it fails once that sounds like a failure